### DISCUSSION TITLE
Improve cloud notification when a dimension switches from collected to not collected frequently

### DISCUSSION THREAD

**Discussion by stelfrag** [netdata/netdata#12224] [Ideas]
Currently for every dimension in a chart we transmit its "liveness" along with the first collected timestamp that the agent knows
about that dimension. 

**Current implementation**
- The chart (and dimension) is considered live after **N** iterations of being collected (wall time would be **N x update_every** defined for the chart)
- The dimension similarly is considered as "stopped being collected" after **N** iterations 

This is problematic (we saw for apps.plugin) for processes that start / stop frequently as they will create and then start / stop collecting dimensions frequently, resulting in constantly sending "dimension updates" to the cloud. 

**Proposal**
- Active charts and dimensions should generate update messages immediately (no need to way **N** iterations)
- Dimensions that stop being collected should wait
  - MAX(chart_obsoletion_time specified, 1.5 x update_every defined for the chart)
- Chart obsoletion by itself will generate messages as needed (all dimensions are gone, chart stopped collecting)
- Database metric rotation will continue to generate messages (first and last data collection timestamps will be 0 to indicate deletion of a dimension)

@amalkov , @ralphm , @ktsaou

**Comment by ktsaou** [comment]
I am in favor of this.

The way it is implemented now, it pushes a lot of messages to the cloud, but the cloud does not get any useful additional info. All the node instances will generate exactly the same messages, so chart routing is not affected by this at all.
